enhancements/sig-architecture/81-addon-lifecycle/README.md

```diff
@@ -37,26 +38,787 @@ proposal's scope.
+- The reason could be `Installing`, `InstallSucceed`, `InstallFailed`, `Upgrading`, `UpgradeSucceed`, `UpgradeFailed`.
+- `lastTransitionTime` record the transition time.   
+
+```yaml
+  - lastTransitionTime: "2022-11-02T05:48:12Z"
+    message: install completed with no errors.
+    reason: Succeed
+    status: "False"
+    type: Progressing
+#  remove condition type Config.
+#  - lastTransitionTime: "2022-11-02T05:48:12Z"
+#    message: install completed with no errors.
+#    reason: Succeed
+#    status: "False"
+#    type: Config
+```
+
+When a new config is updated in the `ManagedClusterAddOn`, the addon manager will first calculate the spec hash and 
+update the `DesiredConfigSpecHash`.
+Then the addon manager will generate `ManifestWork` with annotations `configsSpecHash`.
+
+Next, the addon manager updates the `LastAppliedConfigSpecHash` by checking the `ManifestWork`.
+  - `ManifestWork` annotations `configsSpecHash` matches `DesiredConfigSpecHash`.
+  - `ManifestWork` Condition Available status is true.
+  - `ManifestWork` Condition Available observedGeneration equals to generation.
+  - If it is a fresh install since one addon can have multiple ManifestWorks, the `ManifestWork` condition 
+  ManifestApplied must also be true.
+
+The addon manager then compares the `LastAppliedConfigSpecHash` with `DesiredConfigSpecHash` to update the condition.
+- If `LastAppliedConfigSpecHash` == `DesiredConfigSpecHash`
+  - If `LastAppliedConfigSpecHash` is empty before it equals to `DesiredConfigSpecHash`, update condition Progressing 
+  status to false with reason `InstallSucceed`.
+  - If `LastAppliedConfigSpecHash` is not empty before it equals to `DesiredConfigSpecHash`, update condition Progressing 
+  status to false with reason `UpgradeSucceed`.
+- If `LastAppliedConfigSpecHash` != `DesiredConfigSpecHash`
+  - If `LastAppliedConfigSpecHash` is empty, update condition Progressing status to true with reason `Installing`.
+    - If meets some error during the installing, for example, the `ManifestWork` reports an error, update condition Progressing 
+    status to false with reason `InstallFailed`.
+  - If `LastAppliedConfigSpecHash` is not empty, update condition Progressing status to true with reason `Upgrading`.
+    - If meets some error during the installing, for example, the `ManifestWork` reports an error, update condition Progressing 
+    status to false with reason `UpgradeFailed`.
+
+### Workflow
+Below is an example of `ClusterManagementAddOn`, `ManagedClusterAddOn` and `ManifestWork`:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: AddOnHubConfig
+metadata:
+  name: hub-config-yyy
+spec:
+  desiredVersion: v0.10.0
+```
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  configuration:
+    configs:
+    - group: addon.open-cluster-management.io
+      resource: addondeploymentconfigs
+      name: deploy-config-xxx
```

Comment:
note to self, I think these could linked to the names of AddOnHubConfig in kustomize, but an example proving it is probably important.